modin/pandas/groupby.py

```diff
@@ -987,6 +987,40 @@ def groupby_on_multiple_columns(df, *args, **kwargs):
     ],
 )
 class SeriesGroupBy(DataFrameGroupBy):
+    def __init__(self, df, *args, **kwargs):
+        if not isinstance(df, Series):
+            df = df.squeeze(axis=1)
+        ErrorMessage.catch_bugs_and_request_email(
+            not isinstance(df, Series),
+            f"SeriesGroupBy is supposed to obtain only Series objects, got: {type(df)}",
+        )
+        super(SeriesGroupBy, self).__init__(df.to_frame(), *args, **kwargs)
+        self._name = df.name or "__reduced__"
+
+    _NO_RENAME_METHODS = ["ngroup"]
+
+    def _build_series_wrapper(self, func):
+        def name_setter(*args, **kwargs):
+            if kwargs.get("axis", 0) == 1:
+                raise ValueError("No axis named 1 for object type Series")
+            result = func(*args, **kwargs)
+            if hasattr(result, "_query_compiler"):
+                if not isinstance(result, Series):
+                    result = result.squeeze(axis=1)
```

Comment:
Is it really necessary to do all of this overriding of `__getattribute__` for `groupby.size`? I see a change above that squeezes the result, why is that not enough?

Reply:
that squeeze above is enough for `.size`, but all the other methods don't squeeze their results, and actually they don't have to, since they supposed to be executed under `DataFrame`

Reply:
So is this needed if they do not need to `squeeze`?

Reply:
I meant that `DataFrameGroupbyObj` doesn't need to squeeze its results (all of the methods always return DataFrames), but since the results of `SeriesGroupby` must be Series, it needs to be squeezed.

Reply:
I see, can we have the return object of each method be some property like `_return_type`?

Reply:
Well, it seems that casting Series to Frame brings more troubles than profit. We're just trying to implement logic, that will handle the Series attributes and behaviour, however we can continue to use Series and relief Modin groupby from that responsibility.